src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -217,14 +265,50 @@ public OpenIdConnectAuthenticationHandler(HttpClient backchannel)
 
             message.State = Options.StateDataFormat.Protect(properties);
 
-            var redirectUri = message.CreateAuthenticationRequestUrl();
-            if (!Uri.IsWellFormedUriString(redirectUri, UriKind.Absolute))
+            if (Options.AuthenticationMethod == OpenIdConnectAuthenticationMethod.RedirectGet)
+            {
+                var redirectUri = message.CreateAuthenticationRequestUrl();
+                if (!Uri.IsWellFormedUriString(redirectUri, UriKind.Absolute))
+                {
+                    Logger.LogWarning(Resources.OIDCH_0036_UriIsNotWellFormed, redirectUri);
+                }
+
+                Response.Redirect(redirectUri);
+
+                return true;
+            }
+            else if (Options.AuthenticationMethod == OpenIdConnectAuthenticationMethod.FormPost)
             {
-                Logger.LogWarning(Resources.OIDCH_0036_UriIsNotWellFormed, redirectUri);
+                var inputs = new StringBuilder();
+                foreach (var parameter in message.Parameters)
+                {
+                    var name = Options.HtmlEncoder.HtmlEncode(parameter.Key);
+                    var value = Options.HtmlEncoder.HtmlEncode(parameter.Value);
+
+                    var input = string.Format(CultureInfo.InvariantCulture, InputTagFormat, name, value);
+                    inputs.AppendLine(input);
+                }
+
+                var issuer = Options.HtmlEncoder.HtmlEncode(message.IssuerAddress);
```

Comment:
The issuer address (i.e the authorization or the logout endpoint) is either extracted from the OIDC provider metadata or manually configured by the dev. In both cases, it's not supposed to be URL-encoded.

It can also be replaced from the `RedirectToIdentityProvider` notification but I'm not sure there's something we could do to avoid manual encoding.